manager/controlapi/extension.go

```diff
@@ -13,7 +18,27 @@ import (
 //   or fails validation.
 // - Returns an error if the creation fails.
 func (s *Server) CreateExtension(ctx context.Context, request *api.CreateExtensionRequest) (*api.CreateExtensionResponse, error) {
-	return nil, errors.New("not implemented")
+	if request.Extension == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "extension must be provided")
+	}
```

Comment:
We need to do more validation here. For example, the `ID` should not be filled in, nor should the `Meta` field.